ood-portal-generator/spec/ood_portal_generator_view_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+require File.expand_path '../../lib/ood_portal_generator', __FILE__
+
+describe OodPortalGenerator::View do
+  describe 'dodes config opts match with example opts?' do
+    it 'returns true if opts match' do
+      config_opts = OodPortalGenerator::View.new
+        .instance_variables.map(&:to_s).map { |opt| opt.delete("@") }
+      example_config_opts = File.read(
+          './share/ood_portal_example.yml').scan(/#[\w_]+:/)
+      
+      # delete inst vars that are not actual options in the example file
+      config_opts.delete('protocol')
+      config_opts.delete('oidc_redirect_uri')
+      config_opts.delete('oidc_crypto_passphrase')
+
+      config_opts.map(&:to_s).each do |opt|
+        expect(example_config_opts).to opt_exists?(opt)  
```

Comment:
If you have two arrays of the same string you just need to do a comparison of the two. RSpec should handle the diff if there is a discrepancy well.

    expect(config_opts.sort).to eq(example_config_opts.sort)